overlord/snapstate/snapstate.go

```diff
@@ -1568,13 +1568,23 @@ func doUpdate(ctx context.Context, st *state.State, names []string, updates []mi
 	}
 	var kernelTs, gadgetTs, bootBaseTs *state.TaskSet
 
+	gating, err := HeldSnaps(st)
```

Comment:
is this correct when doUpdate is called by autoRefreshPhase2? anyway in that case this a bit wasteful anyway in that case as what to update or not has been computed already I think?  should we have a flag or pass the map in in the cases this is not relevant?

Reply:
It's correct but unnecessary like you pointed out. I can move the filtering to `updateManyFiltered` since the other callers of `doUpdate` are specific installs or updates